include/zephyr/net/http/server.h

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2023, Emna Rekik
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_HTTP_SERVER_H_
+#define ZEPHYR_INCLUDE_NET_HTTP_SERVER_H_
+
+#include <stdint.h>
+
+#if defined(CONFIG_POSIX_API)
+
+#include <stddef.h>
+#include <poll.h>
+
+#else
+
+#include <zephyr/posix/poll.h>
+
+#endif
+
+#define CLIENT_BUFFER_SIZE         CONFIG_NET_HTTP_SERVER_CLIENT_BUFFER_SIZE
+#define MAX_CLIENTS                CONFIG_NET_HTTP_SERVER_MAX_CLIENTS
+#define MAX_STREAMS                CONFIG_NET_HTTP_SERVER_MAX_STREAMS
+
+#define GET     0x01
+#define POST    0x02
+#define HEAD    0x03
+#define PUT     0x04
+#define DELETE  0x05
```

Comment:
These are already defined in `include/zephyr/net/http/method.h`

Reply:
While trying to remove these, I noticed that these are used in the bitmask. This means that the values are wrong in that case and we cannot use the ones in method.h. I will come up with proper bitmask values for these.